### PR TITLE
DCS-1348 Hooking in search when no identifiers have been provided

### PR DESCRIPTION
**Pull request by andrewrlee**

### CODE REVIEW

File: server/@types/welcome/index.d.ts [hunk 1]
```diff
@@ -45,6 +44,7 @@ declare module 'welcome' {
       fromLocationType: LocationType
       gender?: GenderKeys
       potentialMatches?: PotentialMatch[]
+      isCurrentPrisoner: boolean
```
Comment (andrewrlee) [inline server/@types/welcome/index.d.ts:47]: this was missing but is always present

File: server/routes/bookedtoday/arrivals/searchforexisting/searchForExistingRecordController.ts [hunk 2]
```diff
@@ -21,6 +21,14 @@ export default class SearchForExistingRecordController {
     return data
   }
 
+  public newSearch(): RequestHandler {
+    return async (req, res) => {
+      const { id } = req.params
+      State.searchDetails.clear(res)
+      res.redirect(`/prisoners/${id}/search-for-existing-record`)
+    }
+  }
```
Comment (andrewrlee) [inline server/routes/bookedtoday/arrivals/searchforexisting/searchForExistingRecordController.ts:30]: new endpoint to reset search details on entry

File: server/routes/bookedtoday/choosePrisonerController.ts [hunk 3]
```diff
@@ -13,4 +15,37 @@ export default class ChoosePrisonerController {
...
+         */
+        res.redirect(`/prisoners/${arrival.id}/confirm-arrival`)
+  }
+
+  private handleCurrentPrisoner(arrival: Arrival, res: Response): void | PromiseLike<void> {
+    switch (arrival.fromLocationType) {
+      case LocationType.COURT:
+        return res.redirect(`/prisoners/${arrival.id}/check-court-return`)
+      case LocationType.CUSTODY_SUITE:
+        return res.redirect(`/feature-not-available`)
```
Comment (andrewrlee) [inline server/routes/bookedtoday/choosePrisonerController.ts:36]: this case isn't currently handled

File: server/routes/bookedtoday/choosePrisonerController.ts [hunk 4]
```diff
@@ -13,4 +15,37 @@ export default class ChoosePrisonerController {
       })
     }
   }
+
+  private handleNewPrisoner(arrival: Arrival, res: Response): void | PromiseLike<void> {
+    return !arrival.prisonNumber && !arrival.pncNumber
+      ? res.redirect(`/prisoners/${arrival.id}/search-for-existing-record/new`)
+      : /**
+         * TODO 3 situations to deal with, will redirect to new pages for each of:
+         * - no matches         - show "This person doesn't have a record"
```
Comment (andrewrlee) [inline server/routes/bookedtoday/choosePrisonerController.ts:24]: if an identifier is provided then we assume that the data is accurate (but give an option to bail out)

File: server/views/partials/expectedArrivalCardMacro.njk [hunk 5]
```diff
@@ -12,25 +12,15 @@
...
-
-                            {% elif arrival.fromLocationType === 'PRISON' %}
-                                <a href="/prisoners/{{ arrival.prisonNumber }}/check-transfer" class="govuk-link govuk-link--no-visited-state" data-qa="confirm-arrival"> {{ arrival.lastName }}, {{ arrival.firstName }}</a>
-
+                            {% if user.isReceptionUser %}
+                                {% if arrival.fromLocationType === 'PRISON' %}
+                                    <a href="/prisoners/{{ arrival.prisonNumber }}/check-transfer" class="govuk-link govuk-link--no-visited-state" data-qa="confirm-arrival"> {{ arrival.lastName }}, {{ arrival.firstName }}</a>
+                                {% else %}
+                                    <a href="/confirm-arrival/choose-prisoner/{{ arrival.id }}" class="govuk-link govuk-link--no-visited-state" data-qa="confirm-arrival"> {{ arrival.lastName }}, {{ arrival.firstName }}</a>
+                                {% endif %}
```
Comment (andrewrlee) [inline server/views/partials/expectedArrivalCardMacro.njk:20]: logic moved to controller

**Review by GurnankCheema — APPROVED** [review]
LGTM

File: server/routes/bookedtoday/choosePrisonerController.test.ts [hunk 6]
```diff
@@ -129,3 +133,171 @@ describe('GET /confirm-arrival/choose-prisoner', () => {
...
+          isCurrentPrisoner: false,
+        })
+      )
+      return request(app)
+        .get('/confirm-arrival/choose-prisoner/aaa-111-222')
+        .expect('Content-Type', /text\/plain/)
+        .expect('Location', '/prisoners/1111-2222-3333-4444/search-for-existing-record/new')
+    })
+
+    it('should redirect to search results when not current and PNC provided', () => {
```
Comment (GurnankCheema) [inline server/routes/bookedtoday/choosePrisonerController.test.ts:195]: This test should have PNC number

File: server/routes/bookedtoday/choosePrisonerController.test.ts [hunk 7]
```diff
@@ -129,3 +133,171 @@ describe('GET /confirm-arrival/choose-prisoner', () => {
...
+          pncNumber: undefined,
+          fromLocationType: LocationType.CUSTODY_SUITE,
+          isCurrentPrisoner: false,
+        })
+      )
+      return request(app)
+        .get('/confirm-arrival/choose-prisoner/aaa-111-222')
+        .expect('Content-Type', /text\/plain/)
+        .expect('Location', '/prisoners/1111-2222-3333-4444/confirm-arrival')
+    })
```
Comment (GurnankCheema) [inline server/routes/bookedtoday/choosePrisonerController.test.ts:270]: pnc should be provided

**Pull request merged**